docs/tiltfile_concepts.md

```diff
@@ -144,38 +144,89 @@ can be expanded and investigated.
+resources. For more information, see the [Docker Compose documentation](docker_compose.html).)
+
+### Configuring Kubernetes Resources
 In many cases, Tilt's automatic resource assembly logic will be sufficient for you to run
 your app. However, if you need to configure your Kubernetes resources on top of Tilt's
 automatic assembly, you can do so with a call to [`k8s_resource`](api.html#api.k8s_resource).
-Here we'll discuss the most common configuration arguments: `new_name` and `port_forwards`.
-For discussion of other available arguments, see the [api spec](api.html#api.k8s_resource).
 
-`new_name` allows you to specify a new resource name, in case you do not like the automatically generated one.
+Here we'll discuss the most common configuration options for Kubernetes resources:
+`new_name`, `port_forwards`, and `objects`. For discussion of other available arguments, see
+the [API spec](api.html#api.k8s_resource).
+
+`new_name` allows you to specify a new resource name, in case you do not like the
+automatically generated one:
 
 ```python
 # rename the resource "redis:deployment" to "redis"
-k8s_resource('redis:deployment', new_name='redis')
+k8s_resource(workload='redis:deployment', new_name='redis')
 ```
 
+(Use this pattern to rename individual resources; to programmatically rename all resources,
+see [`workload_to_resource_function`](api.html#api.k8s_resource).)
+
 Tilt also supports a few ways to specify `port_forwards`:
 
 ```python
 # connect localhost:9000 to the default container port
-k8s_resource('frontend', port_forwards=9000)
+k8s_resource(
+  workload='frontend',
+  port_forwards=9000
+)
 
 # connect localhost:9000 to container port 8000
-k8s_resource('frontend', port_forwards='9000:8000')
+k8s_resource(
+  workload='frontend',
+  port_forwards='9000:8000'
+)
 
 # connect localhost:9000 to container port 8000
 # and localhost:9001 to container port 8001
-k8s_resource('frontend', port_forwards=['9000:8000', '9001:8001'])
+k8s_resource(
+  workload='frontend',
+  port_forwards=['9000:8000', '9001:8001']
+)
+```
+
+Additionally, you may want to add additional Kubernetes objects to an existing
+resource (say, group a Secret with the Deployment that makes use of it), or group
+non-workload objects into their own resource (e.g. make a CRD its own resource X,
+so a workload containing an instance of that CR may name X as a dependency). To
+accomplish this, use the `k8s_resource.objects` parameter to specify one or more
+Kubernetes objects.
+
+```python
+# associate an existing Secret and Volume with the "frontend" service
+k8s_resource(
+  workload='frontend',
+  objects=['frontend:secret', 'frontend:volume']
+)
+
+# make a new resource consisting of some objects necessary for cluster setup
+k8s_resource(
+  objects=['my-ns:namespace', 'kafka:crd', 'some-ingress:ingress'],
+  new_name='cluster-setup',
+)
 ```
 
+#### Kubernetes Object ID Syntax
```

Comment:
Maybe

```suggestion
#### Referencing Kubernetes objects
```

Reply:
Also maybe this whole paragraph should reference the concepts that we use in the code/epic: fragment, unique fragment, fulllname, etc

So something like:

To reference k8s objects use a unique fragment. A unique fragment is a prefix of a fullname that is unique among a set of objects. For instance, for foo:deployment:default:apps, unique fragments are:
    foo if there are no other objects with the name foo
    foo:deployment if there are no other deployments named foo
    foo:deployment:default if there are no other deployments named foo in the default namespace
    foo:deployment:default:apps always.
Then go in to more detail about fragments, unique fragments and fullnames

Reply:
I was under the impression that we weren't particularly married to any of that terminology and the concern was consistency over any specific set of verbiage? I took a swing at making the verbiage consist, wdyt?